signal.go

```diff
@@ -48,10 +48,8 @@ func mainInterruptHandler() {
 		isShutdown = true
 		ltndLog.Infof("Shutting down...")
 
-		// Run handlers in LIFO order.
-		for i := range interruptCallbacks {
-			idx := len(interruptCallbacks) - 1 - i
-			callback := interruptCallbacks[idx]
+		// Execute the interrupt callbacks in FIFO order.
+		for _, callback := range interruptCallbacks {
```

Comment:
What's the rationale behind the change to FIFO order from LIFO?

Reply:
So that we can add and execute interrupt handlers as we go, making it work sort of like a `defer`.

Reply:
FWIW defers are executed in LIFO order.